sklearn/cluster/_agglomerative.py

```diff
@@ -891,6 +903,8 @@ def fit(self, X, y=None):
 
         if return_distance:
             self.distances_ = out[-1]
+
+        if self.n_clusters is None:  # distance_threshold was passed
```

Comment:
I think the following is slightly more clear:

```suggestion
        if distance_threshold is not None:
```

Reply:
Fixed in 61e5c5a459d1f4d76912417ee965d11039f82f45.